amethyst_input/src/system.rs

```diff
@@ -58,16 +60,22 @@ where
         Read<'a, EventChannel<Event>>,
         Write<'a, InputHandler<AX, AC>>,
         Write<'a, EventChannel<InputEvent<AC>>>,
+        ReadExpect<'a, ScreenDimensions>,
```

Comment:
Does this _need_ to be `ReadExpect`? Are there _any_ circumstances where this could hurt us? It would be simple to just pass `screen_dimensions.map(|it| it.hidpi_factor()).unwrap_or(1.0)` below, but if there's no way this can be `None` then this is fine.

Reply:
Alright